derive_client/data_types/models.py

```diff
@@ -99,12 +151,13 @@ class ManagerAddress(BaseModel):
     currency: MainnetCurrency | None
 
 
-@dataclass
-class BridgePair:
+@dataclass(config=ConfigDict(arbitrary_types_allowed=True))
+class BridgeContext:
     source_w3: Web3
     target_w3: Web3
     source_token: Contract
-    target_token: Contract
+    source_event: ContractEvent
+    target_event: ContractEvent
```

Comment:
`BridgePair` was no good. While this worked for LayerZero, since the token contracts are also the ones emitting the event, this did not work for Socket Superbridge case, where we need to listen to events from contracts other than the token contracts.